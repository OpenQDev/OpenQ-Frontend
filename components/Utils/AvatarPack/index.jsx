// Third party
import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import ToolTipNew from '../ToolTipNew';

const AvatarPack = ({ avatars }) => {
  return (
    <div className='flex flex-row-reverse relative w-min'>
      {avatars &&
        avatars.map((avatar) => {
          return (
            <div key={avatar.login} className='w-8'>
              <ToolTipNew toolTipText={avatar.login}>
                <div className='h-8 w-8 rounded-full bg-black cursor-pointer overflow-hidden position relative hover:z-10 z-0 hover:border-gray-700 hover:border border border-transparent'>
                  <Link href={avatar.url} target='_blank' rel='noopener noreferrer'>
                    <div data-testid='link'>
                      <Image height={32} width={32} src={avatar.avatarUrl} alt='avatar' />
                    </div>
                  </Link>
                </div>
              </ToolTipNew>
            </div>
          );
        })}
    </div>
  );
};

export default AvatarPack;
