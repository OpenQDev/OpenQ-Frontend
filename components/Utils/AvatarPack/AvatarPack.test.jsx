/**
 * @vi-environment jsdom
 */
import React from 'react';
import AvatarPack from '.';
import renderer from 'react-test-renderer';
// Test cases for full balances, empty balances, and undefined balances.

describe('AvatarPack', () => {
  // Test cases for
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(
      <AvatarPack
        avatars={[
          {
            login: 'OpenQDev',
            url: 'https://github.com/OpenQDev',
            avatarUrl: 'https://avatars.githubusercontent.com/u/77402538?s=200&v=4',
          },
        ]}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
