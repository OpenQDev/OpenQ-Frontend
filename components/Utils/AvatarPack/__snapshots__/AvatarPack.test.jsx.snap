// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarPack > should match DOM Snapshot 1`] = `
<div
  className="flex flex-row-reverse relative w-min"
>
  <div
    className="w-8"
  >
    <div
      className="group undefined"
    >
      <div
        className="h-8 w-8 rounded-full bg-black cursor-pointer overflow-hidden position relative hover:z-10 z-0 hover:border-gray-700 hover:border border border-transparent"
      >
        <a
          href="https://github.com/OpenQDev"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            data-testid="link"
          >
            <img
              alt="avatar"
              data-nimg="1"
              decoding="async"
              height={32}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F77402538%3Fs%3D200%26v%3D4&w=64&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F77402538%3Fs%3D200%26v%3D4&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F77402538%3Fs%3D200%26v%3D4&w=64&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={32}
            />
          </div>
        </a>
      </div>
      <div
        className="justify-center w-full relative hidden z-50 group-hover:block  undefined "
      >
        <div
          className="flex flex-col items-center"
        >
          <div
            className="flex mt-0.5 md:mt-1 tooltip-triangle absolute undefined"
          />
          <div
            className="flex tooltip absolute undefined"
          >
            <div
              className="undefined"
            >
              OpenQDev
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
